contracts/SpeedMarkets/SpeedMarketsAMMResolver.sol

```diff
@@ -0,0 +1,163 @@
+        address _speedMarketsAMM,
+        address _addressManager
+    ) external initializer {
+        setOwner(_owner);
+        speedMarketsAMM = ISpeedMarketsAMM(_speedMarketsAMM);
+        addressManager = IAddressManager(_addressManager);
+        initNonReentrant();
+    }
+
+    /// @notice resolveMarket resolves an active market
+    /// @param market address of the market
+    function resolveMarket(address market, bytes[] calldata priceUpdateData) external payable nonReentrant notPaused {
+        _resolveMarket(market, priceUpdateData);
+    }
+
+    function _resolveMarket(address market, bytes[] memory priceUpdateData) internal {
+        if (!speedMarketsAMM.canResolveMarket(market)) revert CanNotResolve();
+
+        IPyth iPyth = IPyth(addressManager.pyth());
+        bytes32[] memory priceIds = new bytes32[](1);
+        priceIds[0] = speedMarketsAMM.assetToPythId(SpeedMarket(market).asset());
+        uint64 maximumPriceDelayForResolving = speedMarketsAMM.maximumPriceDelayForResolving();
+        PythStructs.PriceFeed[] memory prices = iPyth.parsePriceFeedUpdates{value: iPyth.getUpdateFee(priceUpdateData)}(
+            priceUpdateData,
+            priceIds,
+            SpeedMarket(market).strikeTime(),
+            SpeedMarket(market).strikeTime() + maximumPriceDelayForResolving
+        );
+
+        PythStructs.Price memory price = prices[0].price;
+
+        if (price.price <= 0) revert InvalidPrice();
+
+        speedMarketsAMM.resolveMarketWithPrice(market, price.price);
+    }
+
+    /// @notice resolveMarket resolves an active market with offramp
+    /// @param market address of the market
+    /// @param priceUpdateData price update data
+    /// @param collateral collateral address
+    /// @param toEth whether to offramp to ETH
+    function resolveMarketWithOfframp(
+        address market,
+        bytes[] calldata priceUpdateData,
+        address collateral,
+        bool toEth
+    ) external payable nonReentrant notPaused {
+        if (!speedMarketsAMM.multicollateralEnabled()) revert MulticollateralOnrampDisabled();
+        address user = SpeedMarket(market).user();
+        if (msg.sender != user) revert OnlyMarketOwner();
+        IERC20Upgradeable defaultCollateral = IERC20Upgradeable(SpeedMarket(market).defaultCollateral());
+        if (address(defaultCollateral) != address(speedMarketsAMM.sUSD())) revert InvalidOffRampCollateral();
+        uint amountBefore = speedMarketsAMM.sUSD().balanceOf(user);
+        _resolveMarket(market, priceUpdateData);
+        uint amountDiff = speedMarketsAMM.sUSD().balanceOf(user) - amountBefore;
+        speedMarketsAMM.offrampHelper(user, amountDiff);
+        if (amountDiff > 0) {
+            IMultiCollateralOnOffRamp iMultiCollateralOnOffRamp = IMultiCollateralOnOffRamp(
+                addressManager.multiCollateralOnOffRamp()
+            );
+            if (toEth) {
+                uint offramped = iMultiCollateralOnOffRamp.offrampIntoEth(amountDiff);
```

Comment:
We will have to add the new contract to the Offramp contract as enabled since this is moved away from SpeedMarketsAMM